src/cargo/util/toml/mod.rs

```diff
@@ -480,6 +480,62 @@ impl TomlProfile {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Eq, PartialEq)]
+#[serde(untagged)]
+pub enum StringOrU32 {
+    String(String),
+    U32(u32),
+}
```

Comment:
Let's use the same code for `TomlOptLevel` as well? It already does something similar, but in a slightly different way. Maybe it's better even to go in the opposite direction and reuse that logic here?

Reply:
Hm, or perhaps the best way to rewrite both with `#[serde(deserialize_with="string_or_int")]`?